### PR TITLE
fix: Disable Jetty's webapp configuration

### LINKED ISSUE #6049: Review Jetty startup, disable service loaders
On startup, the Jetty 11 (and pending Jetty 12) server implementation seems to make use of service loaders unnecessarily - we have designed the server to run jakarta servlets (in case we ever migrated to another servlet container), but set up jetty as an embedded server, where we define all of how it is managed. 

We've discovered that when embedding the server in a larger classpath with other Jetty jakarta components, our embedded server is trying to load those components, and causing errors when it does so. We're solving that problem separately, this issue will simply track how to avoid starting more of Jetty's various features than we require.

### PR DESCRIPTION
**Pull request by niloc132**
Simplifies embedded server setup slightly, and avoid calling reflective and ServiceLoader based configuration apis.

Fixes #6049

### CODE REVIEW

File: server/jetty/src/main/java/io/deephaven/server/jetty/JettyBackedGrpcServer.java [hunk 1]
```diff
@@ -93,17 +90,12 @@ public JettyBackedGrpcServer(
         jetty = new Server();
         jetty.addConnector(createConnector(jetty, config));
 
-        final WebAppContext context =
-                new WebAppContext(null, "/", null, null, null, new ErrorPageErrorHandler(), NO_SESSIONS);
-        try {
-            String knownFile = "/ide/index.html";
-            URL ide = JettyBackedGrpcServer.class.getResource(knownFile);
-            Resource jarContents = Resource.newResource(ide.toExternalForm().replace("!" + knownFile, "!/"));
-            context.setBaseResource(ControlledCacheResource.wrap(jarContents));
```
Comment (devinrsmith) [inline server/jetty/src/main/java/io/deephaven/server/jetty/JettyBackedGrpcServer.java:102]: It looks like we don't use `io.deephaven.server.jetty.ControlledCacheResource`; is that correct? If so, should we delete it?

File: server/jetty/src/main/java/io/deephaven/server/jetty/JettyBackedGrpcServer.java [hunk 2]
```diff
@@ -93,17 +90,12 @@ public JettyBackedGrpcServer(
...
-        } catch (IOException ioException) {
-            throw new UncheckedIOException(ioException);
-        }
-        context.setInitParameter(DefaultServlet.CONTEXT_INIT + "dirAllowed", "false");
+        final ServletContextHandler context =
+                new ServletContextHandler(null, "/", null, null, null, new ErrorPageErrorHandler(), NO_SESSIONS);
+        String knownFile = "/ide/index.html";
+        URL ide = JettyBackedGrpcServer.class.getResource(knownFile);
+        String path = ide.toExternalForm().replace("!" + knownFile, "!/");
```
Comment (JamesXNelson) [inline server/jetty/src/main/java/io/deephaven/server/jetty/JettyBackedGrpcServer.java:97]: probably worth a comment about "`"!/"` specifies this resource base is a jar file which should have its contents served" (but worded better maybe)

**Review by JamesXNelson — APPROVED** [review]

File: server/jetty/src/main/java/io/deephaven/server/jetty/JettyBackedGrpcServer.java (hunk 2 above)
Comment (niloc132) [reply server/jetty/src/main/java/io/deephaven/server/jetty/JettyBackedGrpcServer.java]: Reverted to roughly the old code, and added a note.

File: server/jetty/src/main/java/io/deephaven/server/jetty/JettyBackedGrpcServer.java (hunk 1 above)
Comment (niloc132) [reply server/jetty/src/main/java/io/deephaven/server/jetty/JettyBackedGrpcServer.java:102]: Good catch, same problem that js-plugins has... I found that the ServletContextHandler also can be given a base resource and will guide DefaultServlet to provide content from there. We still want to share other settings though - it turns out our DefaultServlet configuration was coming from a jetty xml file, webdefault.xml. Now we explicitly configure it (and don't end up with other annotation scanning from the classpath or service loaders).

**Review by devinrsmith — APPROVED** [review]

**Pull request merged**